src/BaseEnum.php

```diff
@@ -2,40 +2,68 @@
 
 namespace Greg0ire\Enum;
 
-abstract class BaseEnum
+class BaseEnum extends AbstractBaseEnum
 {
-    private static $constCache = array();
+    private static $classes;
+
+    public function __construct($className = null)
+    {
+        if ($className) {
+            $this->addClass($className);
+        }
+    }
 
     /**
-     * Uses reflection to find the constants defined in the class and cache
-     * them in a local property for performance, before returning them.
+     * Add new class|interface name will be used
+     * to find the constants defined
+     *
+     * @param string|array $className the name of the class or interface
+     */
+    public function addClass($className)
```

Comment:
It depends on our class, but inside has a method that tells our class to look for constants in the external class (as well as other interfaces or classes).

By default, when the `getEnumTypes()` is not overriden, then the class should simply inspect itself.

At the moment, the `BaseEnum` class is used exactly like this : we extend it in our client code, stuff constants in the extending class and we are done. Now, what we would like to do is have the possibility to have the constants stored not in the class itself, but in a vendor class that we are not allowed to touch.

All you need to do for this to work is put your code directly in the `BaseEnum` class instead of creating a new one.